### DISCUSSION TITLE
Real time data?

### DISCUSSION THREAD

**Discussion by johnslemmer** [blitz-js/blitz#1529] [Q&A]
Hi,

I started poking around blitz JS a few days ago.  You all have done a lot of work this year. And from what I can see, have created a really nice framework.  Seriously, the zero-api concept is brilliant.  Thanks!

Have you thought about how you might tackle real time data requirements? As an example have you considered how to create say, a real time chat app with Blitz?  I've kind of scoured through some issues and discussions that have skirted around this idea, but haven't found anything too concrete.

I understand that blitz (using react-query under the hood) can feel very reactive.  However, (at least from what I can tell) this is still just polling at a regular interval and not pushed data (via a websocket, Pusher, etc). Polling can be weighty (depending on the size of the query) and it can be worth having a continual connection to push updates over for more real time data needs.

I started thinking about how I could tackle real time data.  And thinking about how I could receive something from Pusher or a websocket and use that to invalidate/update the cache.  Is that something that anyone has tried to do?  Any luck? Is it worth exploring how that might be integrated in to the blitz framework?  There is a [react-query issue discussion](https://github.com/tannerlinsley/react-query/issues/171) around this, but the conclusion was it didn't belong in their library.  Maybe it could belong in blitz?

Thanks, appreciate your thoughts.

**Comment by flybayer** [comment]
Hey @johnslemmer! I haven't personally worked much with apps like this, but Pusher is what I've seen most people using.

As you said, you could simply use pusher to update or invalidate the cache.

If we can come up with a nice pattern that works for various uses cases, then yeah we can see about adding it to blitz.

**Comment by Steffan-Harris** [comment]
I'm currently building an app with real-time multi-channel communication and videos.

I initially tried Twilio, but I'm now experimenting with MessageBird, which has just acquired Pusher, and adds a lot more functionality onto it.

**Reply by emanuallan** [reply]
Hey @scisteffan How'd it go? Also @johnslemmer, did you find anything as well? I am about to begin building a real-time chat app with Blitz, just trying to collect information/suggestions beforehand. Thanks!

**Reply by Steffan-Harris** [reply]
Hey, sorry for the delay in reply! I didn't continue making this app. Hope it goes well for you!

**Reply by mrcampbell** [reply]
I was going to use Apollo's polling/refetch as a cheap way to do it unless I found a better way.

**Reply by mrcampbell** [reply]
Cheap as in, without much effort or LOC.  Most definitely not financially a good idea at scale haha
